channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/dto/JaicpDialerAPI.kt

```diff
@@ -0,0 +1,143 @@
+package com.justai.jaicf.channel.jaicp.dto
+
+import com.justai.jaicf.channel.jaicp.JSON
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonElement
+import java.time.DayOfWeek
+import java.time.Instant
+import java.time.LocalTime
+import java.time.format.DateTimeFormatter
+
+@Serializable
+class JaicpDialerAPI {
```

Comment:
All of the fields and methods of this class are internal or private, why the class is not internal then?
Or maybe vice-versa, why all of its methods are internal? I think, there is nothing to hide

Reply:
It should be internal, i think.
There's nothing to hide, but we provide full class usage from TelephonyReactions. I assume that the less public classes a module provides, the better it is for user.

Reply:
I don't think we should look at the framework core only from the DSL usage side.
A user might want to operate with `RedialData` or `JaicpRedialAPI` programmatically (i.e. not just from DSL).

Reply:
I'm not sure about a use case for it. All constructor fields from RedialData are exposed by `reactions.telephony?.redial`.

Reply:
Yeah, but you are not able to construct `RedialData` and operate with it, do some datetime calculations, store it somewhere, etc. You are not able to operate with CallReportData, serialize it in some other format, print it, etc.
You are only able to do stuff that is explicitly exposed by Reactions, it seems like an unnecessary restriction for the user.

DSL (and any other facade) should **extend** model with some helpers, but it definitely shouldn't **hide** the model behind it.

Reply:
Well, If you need it, you can ctrl-C and ctrl-V this class from source code and write one extension to TelephonyReactions :) 

I found two arguments against exposing these DTOs:
1. RedialData uses Long for timestamp and this is error prone. Reactions API uses Instant, that's better. Putting Instant to DTO requires a custom serialization logic, and I don't like it.
2. I'd like to keep reactions api as minimalistic as possible. For any extra cases there are extension functions and source code. I need a reason (a request from user, for example) to include it into api methods, and i'm not sure that this request will ever come.

Reply:
>  If you need it, you can ctrl-C and ctrl-V this class

That's definitely an unnecessary limitation as I said before.

> RedialData uses Long for timestamp and this is error prone.

We definitely can make `RedialData` public and write as many different constructors, factories and helpers as we want (and as a user wants) and not be limited by one method in Reactions class.

> I'd like to keep reactions api as minimalistic as possible.

The minimalistic view on this shouldn't limit the user. 

> For any extra cases there are extension functions and source code

What an extension I'm supposed to write if I want to operate on `RedialData` **without** actually triggering redial? What an extension I'm supposed to write if I want to access the report? 

I do not really see any motivation to prevent a user from doing that.

Reply:
I forgot about internal constructors, you're right with `RedialData`. It's a complex object and there may be reasons to build it in DSL.

But im still not sure with removing `internal` modified on JaicpDialerAPI class. Just cannot find a reason someone would need it.

Reply:
`JaicpReadialApi` is actually just a response builder. So, the same as for `ReadialData`, why a user is not allowed to build a response programmatically?

Reply:
Because all build functionality is exposed via reactions. Same reason why `JaicpReplies` were internal and sealed.

Reply:
> Same reason why JaicpReplies were internal and sealed.

Yeah, and now we have #113.